core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1525,10 +1539,27 @@ private[spark] class DAGScheduler(
             s" ${task.stageAttemptId} and there is a more recent attempt for that stage " +
             s"(attempt ${failedStage.latestInfo.attemptNumber}) running")
         } else {
+          // Gracefully handling the stage abort due to fetch failure in the
+          // decommission nodes
+          if (!event.reason.asInstanceOf[FetchFailed].countTowardsStageFailures) {
+            // Ignore stage attempts due to fetch failed only
+            // once per attempt
+            if (!failedStage.failedAttemptIds.contains(task.stageAttemptId)) {
+              failedStage.ignoredFailedStageAttempts += 1
+              DecommissionTracker.incrFetchFailIgnoreCnt()
+              failedStage.latestInfo.stageFailureIgnored(true)
+            }
+            logInfo(s"""Ignoring stage failure due to fetch failed from the decommissioned""" +
```

Comment:
should this log message be inside the if statement above. you are just inside a conditional for fetch failure !countTowardsStageFailures. That doesn't mean its from decomissioning.  I haven't seen anything specified here that marks it for decommissioning case

Reply:
Yes, moved the log line inside the if condition